api/pkg/apis/v1alpha1/model/summary.go

```diff
@@ -37,28 +38,22 @@ type SummaryResult struct {
 }
 
 func (s *SummarySpec) UpdateTargetResult(target string, spec TargetResultSpec) {
-	if existing, ok := s.TargetResults[target]; !ok {
+	if v, ok := s.TargetResults[target]; !ok {
 		s.TargetResults[target] = spec
 	} else {
-		status := s.TargetResults[target].Status
+		status := v.Status
 		if spec.Status != "OK" {
 			status = spec.Status
 		}
-		message := s.TargetResults[target].Message
+		message := v.Message
 		if spec.Message != "" {
 			if message != "" {
 				message += "; "
 			}
 			message += spec.Message
 		}
-		updatedResult := TargetResultSpec{
-			Status:           status,
-			Message:          message,
-			ComponentResults: existing.ComponentResults,
-		}
-		for k, v := range spec.ComponentResults {
-			updatedResult.ComponentResults[k] = v
-		}
-		s.TargetResults[target] = updatedResult
```

Comment:
Sorry, I missed this commit, @Haishi2016 , I found you switched to use maps. But seems in this commit, you don't set the updated result back to map. 

We need to add one line to set s.TargetResults[target] = v.